pkg/actuators/machine/instances.go

```diff
@@ -639,3 +645,21 @@ func getInstanceMetadataOptionsRequest(providerConfig *machinev1beta1.AWSMachine
 	}
 	return imdsOptions
 }
+
+func getCapacityReservationSpecification(capacityReservationID string) (*ec2.CapacityReservationSpecification, error) {
+	if capacityReservationID == "" {
+		//  Not targeting any specific Capacity Reservation
+		return nil, nil
+	}
+
+	if !strings.HasPrefix(capacityReservationID, "cr-") {
```

Comment:
Are there any other validations we can apply, maximum length? Valid characters? Is there a regex we could use to validate the user input here?

Reply:
right resource-id will be 17char long excluding `cr-` and we check regex with a mix of `lower alpha` and `digits`.